File: features/favorites/src/main/java/com/vp/favorites/ui/FavoriteActivity.java
package com.vp.favorites;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.ListAdapter;
import androidx.recyclerview.widget.RecyclerView;

import android.content.res.Configuration;
import android.os.Bundle;

import com.vp.core.vm.model.FavouriteMovieModel;

import java.util.ArrayList;
import java.util.List;

public class FavoriteActivity extends AppCompatActivity {

    private RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_favorite);

        bindViews();
        populateMoviesrecyclerview();
    }

    private void bindViews() {
        this.recyclerView = findViewById(R.id.activity_favorite_recycler_view);
    }

    private void populateMoviesrecyclerview() {
        FavouritesMoviesAdapter moviesAdapter = new FavouritesMoviesAdapter(getMoviesFake());
        recyclerView.setAdapter(moviesAdapter);
        recyclerView.setHasFixedSize(true);

        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(layoutManager);
    }

    private List<FavouriteMovieModel> getMoviesFake() {
        List<FavouriteMovieModel> movies = new ArrayList<>();
        movies.add(new FavouriteMovieModel("1", "Title", "Lorem", "Ipsum", "", "", ""));

        return movies;
    }
}

File: features/favorites/src/main/java/com/vp/favorites/ui/FavoriteActivity.kt
package com.vp.favorites.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import com.vp.core.vm.model.FavouriteMovieModel
import com.vp.core.vm.model.ScreenStatus
import com.vp.favorites.R
import com.vp.favorites.ui.adapter.FavouritesMoviesAdapter
import com.vp.favorites.vm.FavoriteViewModel
import kotlinx.android.synthetic.main.activity_favorite.*

class FavoriteActivity : AppCompatActivity() {

    private lateinit var viewModel: FavoriteViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_favorite)

        initViewModel()
        observerFavouritesMovies()
    }

    private fun initViewModel() {
        viewModel = ViewModelProviders.of(this).get(FavoriteViewModel::class.java)
    }

    private fun observerFavouritesMovies() {
        viewModel.screenStatus.observe(this, Observer(::onScreenStateReceived))
    }

    private fun onScreenStateReceived(screenStatus: ScreenStatus) {
        when (screenStatus) {
            is ScreenStatus.LOADING -> showSpinner()
            is ScreenStatus.SUCCESS<*> ->
                showFavouritesMovies(screenStatus.element as List<FavouriteMovieModel>)
        }
    }

    private fun showSpinner() {

    }

    private fun showFavouritesMovies(movies: List<FavouriteMovieModel>) {
        val adapter = FavouritesMoviesAdapter(movies)
        activity_favorite_recycler_view.adapter = adapter
        activity_favorite_recycler_view.setHasFixedSize(true)

        val linearyLayoutManager = LinearLayoutManager(this)
        activity_favorite_recycler_view.layoutManager = linearyLayoutManager
    }
}